pkg/reconciler/eventlistener/resources/container.go

```diff
@@ -70,6 +70,9 @@ func MakeContainer(el *v1beta1.EventListener, configAcc reconcilersource.ConfigA
 	if !cfg.Defaults.IsDefaultRunAsGroupEmpty {
 		containerSecurityContext.RunAsGroup = ptr.Int64(cfg.Defaults.DefaultRunAsGroup)
 	}
+	if *c.SetReadOnlyRootFilesystem {
+		containerSecurityContext.ReadOnlyRootFilesystem = ptr.Bool(true)
+	}
```

Comment:
can we move this under `SetSecurityContext` block itself

I mean if `SetSecurityContext` 

if *c.SetSecurityContext {
                if *c.SetReadOnlyRootFilesystem {
		containerSecurityContext.ReadOnlyRootFilesystem = ptr.Bool(true)
	}
		containerSecurityContext = corev1.SecurityContext{
			AllowPrivilegeEscalation: ptr.Bool(false),
			Capabilities: &corev1.Capabilities{
				Drop: []corev1.Capability{"ALL"},
			},
			RunAsNonRoot: ptr.Bool(cfg.Defaults.DefaultRunAsNonRoot),
			SeccompProfile: &corev1.SeccompProfile{
				Type: corev1.SeccompProfileTypeRuntimeDefault,
			},
		}
	}

Reply:
Fine by me. I will update the flag docs to note that the flag can only be used when el-security-context is set.

Reply:
Updated @savitaashture